CONTRIBUTING.md

```diff
@@ -54,45 +54,75 @@ An essential part of getting your change through is to make sure all existing te
 * Make sure you are at the directory where pom.xml is located.
 
 ##### Unit Tests
-* Run the following:
-
-```
-mvn test
-```
+* Run the following: 
+    ```
+    mvn test
+    ```
 
 ##### Integration Tests
-* By default, the integration tests are not executed. In case you are interested in executing 
-  them, disable the `skipITs` property.
-* The following environment variables are required to run the integration tests. 5, 6, and 7 are
-  only required when running a windows integration test.
-
-1. GOOGLE_PROJECT_ID
-1. GOOGLE_CREDENTIALS
-1. GOOGLE_REGION
-1. GOOGLE_ZONE
-1. GOOGLE_BOOT_DISK_PROJECT_ID
-1. GOOGLE_BOOT_DISK_IMAGE_NAME
-1. GOOGLE_JENKINS_PASSWORD
-1. GOOGLE_SA_NAME
-
-* Run the following:
-```
-mvn verify -DskipITs=false
-```
+
+Integration tests actually provision instances in a GCP project, run pipeline, take snapshot etc.
+Therefore, they are disabled in the CI and expected to be executed by contributors in their laptop itself.
+
+Reasons for disabling integration test in CI,
+* getting provisioning GCP infra is not possible
+* even if we did get a GCP infra setup in the CI, it is risky to expose that, as someone can abuse the CI.
+
+By default, integration tests are skipped from the maven goals, need to enable using the `skipITs` property.
+
+Steps to execute integration test
+* Prepare VM images  
+  (ideally we should automate this see idea [here](https://github.com/jenkinsci/google-compute-engine-plugin/pull/492#discussion_r1892705637))  
+  The jenkins agent images need to have java installed. We have a packer script to create the image and upload to your configured GCP project.
+  The scripts are located in [testimages/linux](./testimages/linux)
+  Navigate to the directory and execute,
+  ```bash
+  bash setup-gce-image.sh
+  ```
+  If you want to execute the integration tests for `non-standard-java` location, then create non-standard-java image as,
```

Comment:
Why would I want to do that? What does this even mean? As a reader new to the project I have no idea what _the integration tests for `non-standard-java` location_ refers to.